zebra/zebra_nhg.c

```diff
@@ -2462,6 +2462,12 @@ static unsigned nexthop_active_check(struct route_node *rn,
 		ifp = if_lookup_by_index(nexthop->ifindex, nexthop->vrf_id);
 
 		if (ifp && (if_is_operative(ifp) || if_is_up(ifp))) {
+			if (family == AFI_IP &&
+			    CHECK_FLAG(nexthop->flags, NEXTHOP_FLAG_ACTIVE) &&
+			    !if_has_connected_routes(ifp, AF_INET)) {
```

Comment:
We should not mark a nexthop as not active if a nexthop is an actual nexthop group in the kernel.

Reply:
> Zebra probably should be putting a NEXTHOP_IFINDEX route back into the kernel if we have detected that the kernel has removed the route

I have changed that